dockertar-sha256-helper.go

```diff
@@ -0,0 +1,32 @@
+package main
+import (
+	"fmt"
+	"compress/gzip"
+	"os"
+	"io"
+	"crypto/sha256"
+)
+
+func main() {
+	reader, err := os.Open(os.Args[1])
+	if err != nil {
+		os.Exit(1)
+	}
+	buf := make([]byte, 4096)
+	sha_256 := sha256.New()
+	w := gzip.NewWriter(sha_256)
+
+	for true {
+		n, err := reader.Read(buf)
+		if err != nil && err != io.EOF {
```

Comment:
nit: Could process the `n > 0` case first as it as suggested in the documentation.
Something like
```
if n > 0 {
  w.Write(buf[:n])
}
if err == io.EOF {
  break
}
if err != nil {
  panic(err)
}